File: data/code_pairs/cpp/1425.cpp
#include <iostream>
#include <cassert>

// Function declaration
int factorial(int n);

// Simple test function declaration
void testFactorial();

int main() {
    testFactorial(); // Run tests

    // Main program logic
    int number;
    std::cout << "Enter a number to calculate factorial: ";
    std::cin >> number;
    std::cout << "The factorial of " << number << " is " << factorial(number) << std::endl;

    return 0;
}

// Function definition
int factorial(int n) {
    int fact = 1;
    for (int i = 1; i <= n; ++i) {
        fact *= i;
    }
    return fact;
}

// Simple test function definition
void testFactorial() {
    assert(factorial(0) == 1); // Test case for factorial of 0
    assert(factorial(1) == 1); // Test case for factorial of 1
    assert(factorial(5) == 120); // Test case for factorial of 5

    std::cout << "All tests passed." << std::endl;
}

File: data/code_pairs/fortran/1425.f90
program mainProgram
  implicit none
  integer :: inputNumber, result

  ! Function declaration
  interface
    function factorial(n) result(fact)
      integer, intent(in) :: n
      integer :: fact
    end function factorial
  end interface

  ! Main program logic
  print *, "Enter a number to calculate factorial: "
  read *, inputNumber
  result = factorial(inputNumber)
  print *, "The factorial of ", inputNumber, " is ", result

contains
  ! Function to calculate factorial
  function factorial(n) result(fact)
    integer, intent(in) :: n
    integer :: fact, i
    fact = 1
    do i = 1, n
      fact = fact * i
    end do
  end function factorial
end program mainProgram